src/main/scala/org/constellation/p2p/Download.scala

```diff
@@ -101,127 +219,79 @@ object Download extends StrictLogging {
       File(dao.snapshotPath, r.snapshot.hash).writeByteArray(KryoSerializer.serializeAnyRef(r))
     }
 
-    def processSnapshotHash(peer: APIClient, hash: String): Boolean = {
-
-      var activePeer = peer
-      var remainingPeers: Seq[APIClient] =
-        peerData.values.map { _.client }.filterNot(_ == activePeer).toSeq
-
-      var done = false
-
-      while (!done && remainingPeers.nonEmpty) {
-
-        val res = Try {
-          activePeer
-            .getBlockingBytesKryo[StoredSnapshot]("storedSnapshot/" + hash, timeout = 100.seconds)
-        }
-        res match {
-          case Failure(e) => e.printStackTrace()
-          case _          =>
-        }
-        res.toOption.foreach { r =>
-          dao.metrics.incrementMetric("downloadedSnapshots")
-          dao.metrics.incrementMetric("snapshotCount")
-          acceptSnapshot(r)
-        }
-        if (res.isFailure) {
-          dao.metrics.incrementMetric("downloadSnapshotDataFailed")
-        }
-        done = res.isSuccess
-
-        if (!done && remainingPeers.nonEmpty) {
-          activePeer = remainingPeers.head
-          remainingPeers = remainingPeers.filterNot(_ == activePeer)
-        }
+    while (!done && remainingPeers.nonEmpty) {
 
+      val res = Try {
+        activePeer
+          .getBlockingBytesKryo[StoredSnapshot]("storedSnapshot/" + hash, timeout = 100.seconds)
       }
+      res match {
+        case Failure(e) => e.printStackTrace()
+        case _          =>
+      }
+      res.toOption.foreach { r =>
+        dao.metrics.incrementMetric("downloadedSnapshots")
```

Comment:
why not just put this in the Success case above?

Reply:
at the beginning of this function I wrote `// TODO` because it's copied 1:1 as it was before and I'll be working on that